src/Kiota.Builder/Writers/Shell/ShellCodeMethodWriter.cs

```diff
@@ -195,9 +213,19 @@ private void WriteCommandHandlerBodyOutput(LanguageWriter writer, CodeMethod ori
 
                 if (typeString != "Stream")
                 {
+                    var formatterOptionsVar = "formatterOptions";
                     writer.WriteLine($"var {formatterVar} = {outputFormatterFactoryParamName}.GetFormatter({outputFormatParamName});");
                     writer.WriteLine($"response = {outputFilterParamName}?.FilterOutput(response, {outputFilterQueryParamName}) ?? response;");
-                    writer.WriteLine($"{formatterVar}.WriteOutput(response);");
+                    writer.WriteLine($"OutputFormatterOptions {formatterOptionsVar} = null;");
```

Comment:
couldn't all that block live in an helper method in commons to avoid having to generate it?